### PR TITLE
Combine PropertyGrid tab information.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Combine tab related data into a record for clarity and safety. Use a List<> instead of an array to avoid complicated insert logic.

Maintain tab selection as an instance instead of an index.

Move error handling to CreateTab so thrown exceptions with messages (missing a bitmap / name for the tab) are actually thrown instead of a silent failure.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6108)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
Arm failed with `MenuStrip_WndProc_InvokeMouseActivateWithHandle_Success`

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs [hunk 1]
```diff
@@ -1666,7 +1647,16 @@ private ToolStripSeparator CreateSeparatorButton()
...
-            PropertyTab tab = CreatePropertyTab(tabType);
+            PropertyTab tab;
+
+            try
+            {
+                tab = CreatePropertyTab(tabType);
+            }
+            catch (Exception)
+            {
+                return null;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:1658]: Would it make sense to `Debug.WriteLine` this exception?

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs [hunk 2]
```diff
@@ -1687,14 +1677,21 @@ private PropertyTab CreateTab(Type tabType, IDesignerHost host)
...
+                        // Just call the default constructor.
+                        tab = (PropertyTab)Activator.CreateInstance(tabType);
+                    }
                 }
-                else
+                catch
                 {
-                    // Just call the default constructor.
-                    tab = (PropertyTab)Activator.CreateInstance(tabType);
+                    tab = null;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:1694]: This seems to be a silent failure. Consider adding either an assert or a `WriteLine`

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good. Is the test failure intermittent?

**Comment by JeremyKuhne** [comment]
> Is the test failure intermittent?

Didn't see any reference to it, just tagging it for now as this is the "first" occurrence.

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:1694]: It was a silent error before. I'll add a writeline.

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/PropertyGrid.cs:1694]: Actually I'll make it an assert.

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**